DOCS: Documentation edits for clarity, usability, and conformation to doc standards

unitALG opened the pull request:
### Description

Based on the following feedback tickets, I edited some of the pages in the manual:

- [DOCATT-8537](https://jira.unity3d.com/browse/DOCATT-8537): Clarified the installation requirements, including adhering to doc standards more (with step-based procedure and adjusted titles). 
- [DOCATT-8578](https://jira.unity3d.com/browse/DOCATT-8578): Merged duplicated information on Control paths between the API docs and the manual. Whenever there was a discrepancy, I used the guidance from the API docs (for example, the manual said that `<layoutName>{usageName}controlName#(displayName)` was the composition of the path, while the InputControlPath class said it was `<Layout>{Usage}#(DisplayName)Name`).
- [DOCATT-8743](https://jira.unity3d.com/browse/DOCATT-8743): Edited several MD files connected to the concepts for bindings, input, and controls, including merging some of the manual & API documentation for InputControl & PlayerInput classes and fixing some links to nonexistent anchors.
- [DOCATT-9373](https://jira.unity3d.com/browse/DOCATT-9373): Removed the incomplete sentence from the InputActionPhase.Canceled enum value doc.

In addition to those content changes, I also performed a number of changes across several pages that were either structural or more global fixes, which I made in separate commits for clarity:

- Added missing UIDs to manual pages: this was in preparation to convert all links using relative paths to use xref/uid values.
- Normalized "NOTE" and "IMPORTANT" admonitions throughout docs because they were extremely inconsistent in terms of styling and presentation, and most of them didn't use the built-in DocFx support for admonitions.
- Applied standards for images, including more consistent sizing and web accessibility-compliant alt text. Also removed unnecessary screen shots of the Editor where possible (to avoid version-specific inconsistencies).
- Removed mini-TOCS that appeared on some (but not all) manual pages. These weren't compliant with the documentation standards and weren't automated, so some were missing sections that had been added later. And DocFx already generates mini-TOCs for the sidebar, so they were redundant.
- Replaced all links using relative paths with xrefs. These were links to pages in the manual and API docs inside the package as well as links to the Engine docs.


### Testing status & QA

- [x] Generated the documentation locally.

### Overall Product Risks

Low risk, since no code was modified.

### Comments to reviewers

- Some of the changes I made affected most or all manual pages, such as replacing relative links with xref-based links, but I made sure to use a separate commit to make it easier to see the changes that were content-only.
- Any time I made changes to headings I updated any links referring to those headings.

### Checklist

- [x] No changelog entry added..
- [x] Commit message for squash-merge is prefixed with `DOCS: `.
- [x] No forward/backward port needed.

unitALG committed:
Added missing UIDs to manual pages

unitALG committed:
Normalized "NOTE" and "IMPORTANT" admonitions throughout docs

- Content was not modified: this was only to make everything consistent and leverage the extended Markdown feature

unitALG committed:
Merge branch 'develop' into docatt-manual-page-edits

# Conflicts:
#	Packages/com.unity.inputsystem/Documentation~/PlayerInput.md

unitALG committed:
Fixed two more irregular admonitions in ActionBindings.md

unitALG committed:
Image fixes for consistency and web accessibility

- Applied consistent sizing for most component windows
- Updated the alt text to comply with accessibility standards (not an exhaustive search; just the ones I happened to find)
- Removed some images that were redundant or purely screen shots describing the UI elements that appear in the Editor (future-proofing)

unitALG committed:
DOCATT-9373: Removed the incomplete sentence in the Canceled property

- Also replaced "see <reference>" with "refer to <>" in compliance with style guide guidance

unitALG committed:
DOCATT-8578: Merged repeated information on Control paths

- Similar data but somewhat conflicting information on structure & syntax for InputControl paths
- Merged the information from API to Manual topic on the subject, preferring the API information when it diverged
- Also restructured the table a little to include links to related APIs and manual topics instead of examples
- Removed only the control path information from the InputControlPath class and left a link to the manual topic behind

unitALG committed:
DOCATT-8537: Clarified the installation requirements

- Updated the installation instructions to use steps and make it standard compliant
- Also changed the titles to be doc standard compliant (and updated external links pointing there)

unitALG committed:
Removed redundant mini-TOCs that appear on random pages

- These weren't doc standard compliant and weren't automated (some were missing added sections)
- The sidebar TOCs are already automatically generated by DocFx

unitALG committed:
Bulk-replaced all relative paths and links to Engine docs with xrefs

unitALG committed:
DOCATT-8743: Edited PlayerInput class for clarity, made more explicit

- Similar to DOCATT-8578, I merged some of the manual & API documentation for InputControl & PlayerInput classes as part of edits
- Also corrected some links to anchors that didn't exist in Sensors.md, HID.md, and Testing.md

unitALG committed:
Ran reformatting tool

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

# Conflicts:
#	Packages/com.unity.inputsystem/Documentation~/ActionBindings.md

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/ActionAssets.md:
```diff
@@ -3,27 +3,22 @@ uid: input-system-action-assets
-An Input Action Asset is an Asset which contains a set of [Input Actions](Actions.md) definitions and their associated [Bindings](ActionBindings.md) and [Control Schemes](ActionBindings.md#control-schemes). These Assets have the `.inputactions` file extension and are stored in a plain JSON format.
-
-The input system creates an Action Asset when you set up the [default project-wide actions](ProjectWideActions.md), but you can also create new Action Assets directly in the Project window.
+The input system creates an Action Asset when you set up the [default project-wide actions](xref:project-wide-actions), but you can also create new Action Assets directly in the Project window.
```
We say "Action Asset" in this paragraph but "Input Action Asset" in the previous paragraph, shouldn't they be consistent?

unitALG replied:
Yes, absolutely. Unfortunately, I didn't have time to search for consistencies and apply our standards to all MD files, just a handful. The only changes to this file that I made were to switch the links to use xrefs, remove the mini-TOCs at the beginning and to change the Note blocks to use DocFx standard components.

But since you flagged it, I will make another pass on the file to apply consistency.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/ActionAssets.md:
```diff
@@ -3,27 +3,22 @@ uid: input-system-action-assets
 To bring up the Action editor, double-click an `.inputactions` Asset in the Project Browser, or select the __Edit Asset__ button in the Inspector for that Asset. You can have more than one editor window open at the same time, but not for the same Asset.
 
-The Actions Editor which opens is identical to the [Actions Editor in the Project Settings window](ActionsEditor.md).
+The Actions Editor which opens is identical to the [Actions Editor in the Project Settings window](xref:input-system-configuring-input).
```
Here we call it Actions Editor again. Just looking for what they should be called for consistency.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Actions.md:
```diff
@@ -33,36 +33,36 @@ When scripting with Actions in the Input System, there are number of important A
 ### Creating Actions using the Action editor
 
-For information on how to create and edit Input Actions in the editor, see the [Input Actions editor](ActionsEditor.md). This is the recommended workflow if you want to organise all your input actions and bindings in one place, which applies across the whole of your project. This often the case for most types of game or app.
+For information on how to create and edit Input Actions in the editor, see the [Input Actions editor](xref:input-system-configuring-input). This is the recommended workflow if you want to organise all your input actions and bindings in one place, which applies across the whole of your project. This often the case for most types of game or app.
```
Typo: "This often the case....", another typo: ".....most types of game or app"

unitALG replied:
I'll take another pass at this file too.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Actions.md:
```diff
@@ -77,7 +77,7 @@ public class ExampleScript : MonoBehaviour
 The result is similar to using an Actions defined in the Input Actions editor, except the Actions are defined in the GameObject's properties and saved as Scene or Prefab data, instead of in a dedicated Asset.
 
-When you embed actions like this, by defining serialized InputAction fields in a MonoBehaviour, the GameObject's Inspector window displays an interface similar to the Actions column of the [Actions Editor](./ActionsEditor.md), which allows you to set up the bindings for those actions. For example:
+When you embed actions like this, by defining serialized InputAction fields in a MonoBehaviour, the GameObject's Inspector window displays an interface similar to the Actions column of the [Actions Editor](xref:input-system-configuring-input), which allows you to set up the bindings for those actions. For example:
```
Should it be `InputAction` with code formatting or even link?
Should it be `MonoBehaviour` with code formatting or even link?

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Actions.md:
```diff
@@ -151,8 +151,8 @@ lookAction.Enable();
+When you enable an Action, the Input System resolves its bindings, unless it has done so already, or if the set of devices that the Action can use has not changed. For more details about this process, see the documentation on [binding resolution](xref:input-system-action-bindings#binding-resolution).
 
-You can't change certain aspects of the configuration, such Action Bindings, while an Action is enabled. To stop Actions or Action Maps from responding to input, call  [`Disable`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_Disable).
+You can't change certain aspects of the configuration, such Action Bindings, while an Action is enabled. To stop Actions or Action Maps from responding to input, call  [`Disable`](xref:UnityEngine.InputSystem.InputAction.Disable).
```
Typo "such Action Bindings"

ekcoh reviewed (commented):
Doing partial reviews of this PR before approving to provide partial feedback.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Controls.md:
```diff
@@ -185,101 +195,118 @@ history.StartRecording();
 
-- [Interactive rebinding](ActionBindings.md#interactive-rebinding) (`InputActionRebindingExceptions.RebindOperation`) uses it to select between multiple suitable Controls to find the one that is actuated the most.
-- [Conflict resolution](ActionBindings.md#conflicting-inputs) between multiple Controls that are bound to the same action uses it to decide which Control gets to drive the action.
+For example, the PS4 controller has a gyroscope sensor built into the device which constantly feeds data about the angular velocity of the device, even if the device just sits there without user interaction. Conversely, the controller's sticks and buttons require user interaction to produce non-default values.
```
The latter part of this paragraph isn't true. The most general reason for noise in a signal is electric noise in the sensor itself. For devices where this is not filtered in firmware, driver, or OS, such controls are always noisy. E.g. gamepad sticks always produce some noise unless already filtered when they are reported to the Input System. Hence, why sticks have a dead-zone filter by default.

unitALG replied:
I got that explanation directly from the [InputControl.noisy](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.15/api/UnityEngine.InputSystem.InputControl.html#UnityEngine_InputSystem_InputControl_noisy) documentation. Did I misunderstand/misinterpret it or is the existing documentation incorrect?

ekcoh reviewed (commented):
Here is another batch of feedback.  Will resume from Debugging.md tomorrow.

ekcoh committed:
Merge branch 'develop' into docatt-substantial-doc-edits

ekcoh commented:
@unitALG I merged develop into this branch since it was outdated.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Gamepad.md:
```diff
@@ -98,15 +89,15 @@ You can do the same in your C# state structs.
 On Windows (XInput controllers only), Universal Windows Platform (UWP), and Switch, Unity polls gamepads explicitly rather than deliver updates as events.
 
-You can control polling frequency manually. The default polling frequency is 60 Hz. Use [`InputSystem.pollingFrequency`](../api/UnityEngine.InputSystem.InputSystem.html#UnityEngine_InputSystem_InputSystem_pollingFrequency) to get or set the frequency.
+You can control polling frequency manually. The default polling frequency is 60 Hz. Use [`InputSystem.pollingFrequency`](xref:UnityEngine.InputSystem.InputSystem.pollingFrequency) to get or set the frequency.
```
I realise this is currently missing a recent change we have done in this area. Hence I would suggest we change this to something like "The default polling frequency is set by the platform to provide a good user experience for the devices supported on the platform. This frequency is guaranteed to be at least 60 Hz. You may override the polling frequency suggested by the target platform by setting InputSystem.pollingFrequency explicitly in run-time."

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Gamepad.md:
```diff
@@ -149,57 +141,59 @@ InputSystem.ResumeHaptics();
 ## PlayStation controllers
 
-PlayStation controllers are well supported on different Devices. The Input System implements these as different derived types of the [`DualShockGamepad`](../api/UnityEngine.InputSystem.DualShock.DualShockGamepad.html) base class, which derives from [`Gamepad`](../api/UnityEngine.InputSystem.Gamepad.html)):
+PlayStation controllers are well supported on different Devices. The Input System implements these as different derived types of the [`DualShockGamepad`](xref:UnityEngine.InputSystem.DualShock.DualShockGamepad) base class, which derives from [`Gamepad`](xref:UnityEngine.InputSystem.Gamepad)):
```
Typo? I do not think we should have capitalised Devices here? We are talking about "devices" as in "target device" (platform) and not input devices.

unitALG replied:
You're right. I've replaced the instances of Device and Control outside of API or proper names.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Gamepad.md:
```diff
@@ -149,57 +141,59 @@ InputSystem.ResumeHaptics();
+* [`SetLightBarColor(Color)`](xref:UnityEngine.InputSystem.DualShock.DualShockGamepad.SetLightBarColor(UnityEngine.Color)): Used to set the color of the light bar on the controller.
 
-Note that, due to limitations in the USB driver and/or the hardware, only one IOCTL (input/output control) command can be serviced at a time. [`SetLightBarColor(Color)`](../api/UnityEngine.InputSystem.DualShock.DualShockGamepad.html#UnityEngine_InputSystem_DualShock_DualShockGamepad_SetLightBarColor_UnityEngine_Color_) and [`SetMotorSpeeds(Single, Single)`](../api/UnityEngine.InputSystem.Gamepad. ...
+Note that, due to limitations in the USB driver and/or the hardware, only one IOCTL (input/output control) command can be serviced at a time. [`SetLightBarColor(Color)`](xref:UnityEngine.InputSystem.DualShock.DualShockGamepad.SetLightBarColor(UnityEngine.Color)) and [`SetMotorSpeeds(Single, Single)`](xref:UnityEngine.InputSystem.Gamepad.SetMotorSpeeds(System.Single,System.Single)) functionality o ...
```
This paragraph contains some disinformation, since it is not introduced in this PR we can leave it for now and handle it outside this PR but wanted to point this out.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Gamepad.md:
```diff
@@ -149,57 +141,59 @@ InputSystem.ResumeHaptics();
 ## Xbox controllers
 
-Xbox controllers are well supported on different Devices. The Input System implements these using the [`XInputController`](../api/UnityEngine.InputSystem.XInput.XInputController.html) class, which derives from [`Gamepad`](../api/UnityEngine.InputSystem.Gamepad.html). On Windows and UWP, Unity uses the XInput API to connect to any type of supported XInput controller, including all Xbox One or Xbox ...
+Xbox controllers are well supported on different Devices. The Input System implements these using the [`XInputController`](xref:UnityEngine.InputSystem.XInput.XInputController) class, which derives from [`Gamepad`](xref:UnityEngine.InputSystem.Gamepad). On Windows and UWP, Unity uses the XInput API to connect to any type of supported XInput controller, including all Xbox One or Xbox 360-compatibl ...
```
Also here it looks like we have capitalised "Devices" without clear reason?

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/HID.md:
```diff
@@ -129,13 +129,14 @@ This example workflow uses the same technique as the previous example, but provi
 
->__Tip__: If you don't know the format of a given HID you want to support, you can open the Input Debugger with the Device plugged in and pop up both the debugger view for the Device and the window showing the HID descriptor. Then, you can go through the Controls one by one, see what happens in the debug view, and correlate that to the Controls in the HID descriptor. You can also double-click ind ...
+> [!TIP]
+> If you don't know the format of a given HID you want to support, you can open the Input Debugger with the Device plugged in and pop up both the debugger view for the Device and the window showing the HID descriptor. Then, you can go through the Controls one by one, see what happens in the debug view, and correlate that to the Controls in the HID descriptor. You can also double-click individual  ...
```
Not sure if it is part of language guidelines but "pop up" sounds a bit too relaxed and unclear compared to "open" or similar?

unitALG replied:
Indeed. I think "open" should work better.

unitALG replied:
I also noticed that Device and Control was capitalized in this file so I've replaced them where relevant.

ekcoh reviewed (commented):
I have now reviewed up until timing-missed-duplicate-events.md so will resume from there later.

ekcoh reviewed (approved):
Thanks for the comprehensive update. Will see fi I can help with xmldoc validation failures seen in CI. Added some minor comments on formulation and content to consider before landing this.

ekcoh committed:
Update PlayerInput.cs

Removed <para> that seemed accidental causing package validation error.

ekcoh commented:
Attempted to fix package validation error for PlayerInput.cs in #a02dd90.

ekcoh commented:
I currently do not see what might be causing "
PVP-150-1: UnityEngine.InputSystem.InputControl: synthetic: text or XML content outside a top-level tag" in InputControl.cs. Maybe @jfreire-unity or @unitALG  see something I do not on that validation error.

jfreire-unity commented:
There seems to be a lot of errors for images that are not being used in the docs anymore?
Also, some errors due to NDA platforms keywords being present, it seems. Not sure if we need to add validation exceptions for it. If you go through the Yamato failures, it seems pretty explicit.
I'll leave it up for @unitALG to act on it meanwhile.

ekcoh committed:
Merge branch 'develop' into docatt-substantial-doc-edits

ekcoh commented:
Merged develop into this branch to get critical CI fix from https://github.com/Unity-Technologies/InputSystem/pull/2260. Hope this doesn't caused any inconvenience for you.

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Follow-up edits for consistency after reviews

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Re-formatted changes in code samples in MD files

unitALG committed:
Fixed typos and reorganized sections in Input Actions Editor page

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Fixed failing tests for images:

- Removed GeneratePrecompiledLayout.png which was genuinely not referenced anywhere
- The rest are all still actually being used in the docs, but likely were missed because they were sitting inside an HTML <img> tag instead of the usual Markdown structure, so I switched to use the Markdown structure.

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Attempted to fix link errors reported by Windows functional tests

unitALG committed:
Test whether the NDA validation still fails with a relative path

- Switched back to '../api/UnityEngine.InputSystem.Switch.SwitchProControllerHID.html' from 'xref:UnityEngine.InputSystem.Switch.SwitchProControllerHID'

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Fixed missing xref: in link

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Address failures in PVP-150-1

unitALG committed:
Fix Xmldoc issue (para tag doesn't work; use pure MD instead)

unitALG committed:
Fix <summary> tag for isSinglePlayer

unitALG committed:
Try to fix InputTestFixture remarks (code sample is mangled)

unitALG committed:
Removed links to methods in InputTestFixture (Windows can't find them)

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Gave up on xref links and switched back to absolute hard-coded URLs

unitALG committed:
More xref links back to absolute hard-coded URLs

unitALG committed:
Yet more xref links back to absolute hard-coded URLs

unitALG committed:
Next page of xref reversions

unitALG committed:
More xref reversions

unitALG committed:
Removed the rest of the xref links to the Editor Core docs

unitALG committed:
More reversions

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits

unitALG committed:
Reversions up to UISupport.md

unitALG committed:
Replaced "do" with "use" in UISupport.md

unitALG committed:
Merge branch 'develop' into docatt-substantial-doc-edits
